core/vpn/src/network.rs

```diff
@@ -780,6 +662,40 @@ impl ArbiterExt for Arbiter {
     }
 }
 
+fn pick_random_ethernet_addr() -> EthernetAddress {
+    loop {
+        let addr = EthernetAddress(rand::random());
+        if addr.is_unicast() {
+            break addr;
+        }
+    }
```

Comment:
Alternatively, you could use the IP address to construct the mac address (i.e. choose the first 2 bytes, then assign the last 4 bytes from an ipv4 address like here https://github.com/golemfactory/ya-runtime-vm/blob/master/runtime/init-container/src/network.c#L262-L267). This would make the whole network setup more predictable